openquake/hazardlib/source/base.py

```diff
@@ -40,6 +40,8 @@ class BaseSeismicSource(with_metaclass(abc.ABCMeta)):
     __slots__ = ['source_id', 'name', 'tectonic_region_type',
                  'trt_model_id', 'weight', 'seed', 'id']
 
+    MODIFICATIONS = abc.abstractproperty()
```

Comment:
Again, this follows the example set in the MFD base clase, which was put in by Anton and Damiano - long before I got my hands on it. I think eventually all of the source classes may have some sort of modification method attached to them (even if that is not currently the case for the area or point source), so my feeling at this point is that forcing the specification of the attribute might be desirable in order to force the developer to consider the potential modifcations to the class within the logic tree. But, again, I do not have a strong opinion on this, so please advise as to which approach you think is "cleaner/better/tidier/sexier" at present.